src/PostgREST/Plan.hs

```diff
@@ -363,6 +364,21 @@ mutatePlan mutation qi ApiRequest{..} sCache readReq = mapLeft ApiRequestError $
     rootOrder = maybe [] snd $ find (\(x, _) -> null x) qsOrder
     combinedLogic = foldr addFilterToLogicForest logic qsFiltersRoot
     body = payRaw <$> iPayload -- the body is assumed to be json at this stage(ApiRequest validates)
+    typedColumnsOrError = mapStopOnLeft (resolveOrError (HM.lookup qi $ dbTables sCache)) $ S.toList iColumns
+
+resolveOrError :: Maybe Table -> FieldName -> Either ApiRequestError TypedField
+resolveOrError Nothing _ = Left NotFound
+resolveOrError (Just table) field =
+  case resolveTableField table field of
+    Nothing         -> Left $ ColumnNotFound field
+    Just typedField -> Right typedField
+
+mapStopOnLeft :: (a -> Either b c) -> [a] -> Either b [c]
+mapStopOnLeft _ [] = Right []
+mapStopOnLeft f (x:xs) =
+  case f x of
+    Left e  -> Left e
+    Right y -> (y:) <$> mapStopOnLeft f xs
```

Comment:
This looks like [traverse](https://hackage.haskell.org/package/base-4.17.0.0/docs/Prelude.html#v:traverse)

> (a -> f b) -> t a -> f (t b)

You might be able to use it like 

```haskell
typedColumnsOrError = resolveOrError (HM.lookup qi $ dbTables sCache) `traverse` S.toList iColumns
```

---

There is an example on https://github.com/PostgREST/postgrest/blob/78d45b4e32e1129f25c2b6739dbd587ef873822c/src/PostgREST/Plan.hs#L310

Reply:
Oh. Newbie mistake. `traverse` makes perfect sense, thanks!

(As a non-Haskell guy, it's not always obvious to me what's implicit in the behaviours of various types. Like in this case how the Functor and Applicative implementations for `Either` lead to a nice short-circuit behaviour when traversing and hitting that first `Left`.)